examples/block_writer/main.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy
+
+// The block writer example program is a write-only workload intended to insert
+// a large amount of data into cockroach quickly. This example is intended to
+// trigger range splits and rebalances.
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"math/rand"
+	"net/url"
+	"sync/atomic"
+	"time"
+
+	"github.com/cockroachdb/cockroach/security"
+	"github.com/cockroachdb/cockroach/security/securitytest"
+	"github.com/cockroachdb/cockroach/server"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/log"
+	"github.com/cockroachdb/cockroach/util/uuid"
+)
+
+const (
+	insertBlockStmt = `INSERT INTO blocks (writer_id, block_id, raw_bytes) VALUES ($1, $2, $3)`
+)
+
+// db-url = URL of database. If none is specified, a local cluster is
+// automatically started.
+var dbURL = flag.String("db-url", "", "URL to connect to a running cockroach cluster.")
+
+// concurrency = number of concurrent insertion processes.
+var concurrency = flag.Int("concurrency", 3, "Number of concurrent writers inserting money.")
+
+// outputInterval = interval at which information is output to console.
+var outputInterval = flag.Duration("output-interval", 1*time.Second, "Interval of output.")
+
+// Minimum and maximum size of inserted blocks.
+var minBlockSizeBytes = flag.Int("min-block-bytes", 256, "Minimum amount of raw data written with each insertion.")
+var maxBlockSizeBytes = flag.Int("max-block-bytes", 1024, "Maximum amount of raw data written with each insertion.")
+
+// numBlocks keeps a global count of successfully written blocks.
+var numBlocks uint64
+
+// A blockWriter writes blocks of random data into cockroach in an infinite
+// loop.
+type blockWriter struct {
+	id   string
+	db   *sql.DB
+	rand *rand.Rand
+}
+
+func newBlockWriter(db *sql.DB) blockWriter {
+	source := rand.NewSource(int64(time.Now().UnixNano()))
+	return blockWriter{
+		db:   db,
+		id:   uuid.NewUUID4().String(),
+		rand: rand.New(source),
+	}
+}
+
+// run is an infinite loop in which the blockWriter continuously attempts to
+// write blocks of random data into a table in cockroach DB.
+func (bw blockWriter) run() {
+	for {
+		blockID := bw.rand.Int63()
```

Comment:
It's random because I intentionally want it to randomly distribute across the primary keyspace